frontend/src/modules/Dashboard/Components/GroupCard.tsx

```diff
@@ -37,32 +41,108 @@ export const GroupCard = ({
     }
     setOpen(false)
   }
+  // returns color of background, button, and if newly matchable
+  function getColor(students: number, groups: number) {
+    //all students are matched
+    if (students === 0 && groups > 0) {
+      return [colors.paleviolet, colors.darkpurple, 'no']
+    }
+    //students are ready to be matched
+    else if (newStudents > 0 && groupsFormed > 0) {
+      return [colors.lightgreen, colors.darkgreen, 'no']
+    }
+    //NEWLY MATCHABLE
+    else if (newStudents > 1 && groupsFormed === 0) {
+      return [colors.lightgreen, colors.darkgreen, 'yes']
+    }
+    //only 1 student & 0 groups formed
+    else return [colors.yellow, colors.yellow, 'no']
+  }
+  const styleArray = getColor(newStudents, groupsFormed)
```

Comment:
I don't see `styleArray[1]` being used anywhere in the latest version of this PR. Suggest removing the middle array element. Also `styleArray[2]` is only being used in one place